File: 5585.cpp
#include <iostream>
#include <array>
using namespace std;

constexpr int PAY = 1'000;

constexpr array<int, 6> coins {
    500, 100, 50, 10, 5, 1
};

int main() {
    cin.tie(nullptr)->sync_with_stdio(false);
    int price;
    cin >> price;
    int change = PAY - price;

    int coins_count = 0;
    while (change != 0) {
        for (const auto &c : coins) {
            int diff = change - c;
            if (diff < 0) continue;
            change = diff;
            coins_count += 1;
            break;
        }
    }

    cout << coins_count << '\n';
}

File: 5585.rs
use std::io::stdin;

const PAY: i32 = 1_000;
const COINS: [i32; 6] = [500, 100, 50, 10, 5, 1];

fn main() {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    let input: i32 = input.trim().parse().unwrap();

    let mut change = PAY - input;
    let mut coins_count = 0;

    while change > 0 {
        for coin in COINS {
            let diff = change - coin;
            if diff < 0 {
                continue;
            }
            change = diff;
            coins_count += 1;
            break;
        }
    }

    println!("{}", coins_count);

}
